src/Stratis.Features.Unity3dApi/LocalCallContract.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections.Generic;
+using System;
+using System.Linq;
+using NBitcoin;
+using Stratis.Bitcoin.Features.SmartContracts.Models;
+using Stratis.Bitcoin.Features.SmartContracts.Wallet;
+using Stratis.Bitcoin.Features.SmartContracts;
+using Stratis.SmartContracts.CLR.Caching;
+using Stratis.SmartContracts.CLR.Local;
+using Stratis.SmartContracts.CLR.Serialization;
+using Stratis.SmartContracts.CLR;
+using Stratis.SmartContracts;
+
+namespace Stratis.Features.Unity3dApi
+{
+    // TODO: Move this to a more central point once 1.4.0.0 stabilises
+    public interface ILocalCallContract
+    {
+        LocalExecutionResponse LocalCallSmartContract(LocalCallContractRequest request);
+        T LocalCallSmartContract<T>(ulong? blockHeight, string sender, string contractAddress, string methodName, params object[] arguments);
+    }
+
+    public class LocalCallContract : ILocalCallContract
+    {
+        private readonly Network network;
+        private readonly ChainIndexer chainIndexer;
+        private readonly ISmartContractTransactionService smartContractTransactionService;
+        private readonly ILocalExecutor localExecutor;
+        private readonly IContractPrimitiveSerializer primitiveSerializer;
+        private readonly IContractAssemblyCache contractAssemblyCache;
+
+        public LocalCallContract(Network network, ISmartContractTransactionService smartContractTransactionService, ChainIndexer chainIndexer, ILocalExecutor localExecutor)
+        {
+            this.network = network;
+            this.chainIndexer = chainIndexer;
+            this.smartContractTransactionService = smartContractTransactionService;
+            this.localExecutor = localExecutor;
+        }
+
+        public LocalExecutionResponse LocalCallSmartContract(LocalCallContractRequest request)
+        {
+            ContractTxData txData = this.smartContractTransactionService.BuildLocalCallTxData(request);
+
+            var height = request.BlockHeight ?? (ulong)this.chainIndexer.Height;
+
+            ILocalExecutionResult result = this.localExecutor.Execute(
+                height,
+                request.Sender?.ToUint160(this.network) ?? new uint160(),
```

Comment:
Is `Sender` not required?

Reply:
For a local call it seems to be almost unimportant what the actual sender address is. This class is actually copied from one of @quantumagi's pull requests, so any changes would also need to be reflected there too